cmake/Hipify.cmake

```diff
@@ -1,11 +1,54 @@
 # cmake file to trigger hipify
 
+set(DICT_FILE ${CMAKE_BINARY_DIR}/hipify_output_dict_dump.txt)
+
 # Get the hipify_cli.py source directory from current directory by going to parent directory
-get_filename_component(HIPIFY_SCRIPTS_DIR ${CMAKE_CURRENT_LIST_DIR} DIRECTORY)
+get_filename_component(HIPIFY_DIR ${CMAKE_CURRENT_LIST_DIR} DIRECTORY)
+
+function(write_file_list FILE_SUFFIX INPUT_LIST)
+  message(STATUS "Writing ${FILE_SUFFIX} into file - file_${FILE_SUFFIX}.txt")
+  set(_FULL_FILE_NAME "${CMAKE_BINARY_DIR}/cuda_to_hip_list_${FILE_SUFFIX}.txt")
+  file(WRITE ${_FULL_FILE_NAME} "")
+  foreach(_SOURCE_FILE ${INPUT_LIST})
+    file(APPEND ${_FULL_FILE_NAME} ${CMAKE_CURRENT_SOURCE_DIR}/${_SOURCE_FILE})
```

Comment:
Since hipify script is triggered only once for the whole project it uses - ${project_source_dir} = /home/pruthvi/work/tp_rocm

When - write_file_list() is called it is triggered in each CMakeLists.txt context, where - ${CMAKE_CURRENT_SOURCE_DIR} = /home/pruthvi/work/tp_rocm/tensorpipe
and this info is required to update the correct file paths.

Reply:
Noting here for discussion: would CMAKE_CURRENT_LIST_DIR or CMAKE_CURRENT_SOURCE_DIR be more appropriate in the general scenario?

Reply:
CMAKE_CURRENT_LIST_DIR will point to the directory where this Hipify.cmake resides. We don't want that; we want the project CMakeLists.txt file path.